### PR TITLE
Show groups for downloads / installs

### PR DESCRIPTION
**Pull request by corymcdonald**
## Show groups for downloads / installs
Screen Shot 2019-12-06 at 3 50 16 PM
Screen Shot 2019-12-06 at 3 50 25 PM
Screen Shot 2019-12-06 at 3 50 21 PM

#### Features

- Only have one `promo_registrations_controller`
- Single source of truth for the referral panel
- Show rate cards for different groups

#### How To Test

This is dependent on the following PR from Promo Server
https://github.com/brave-intl/vault-promo-services/pull/121

### CODE REVIEW

File: app/javascript/packs/referrals/ReferralPanel.tsx [hunk 1]
```diff
@@ -7,34 +7,50 @@ import en, { flattenMessages } from "../../locale/en";
...
 import Information from "./Information";
 
 import routes from "../../views/routes";
 
+import ArrowPointer from "./referralPanel/ArrowPointer";
+import Groups from "./referralPanel/Groups";
+import Stats from "./referralPanel/Stats";
+
+export enum ReferralType {
+  Finalized = "finalized",
```
Comment (yachtcaptain23) [inline app/javascript/packs/referrals/ReferralPanel.tsx]: I believe this should be all caps. E.g. https://google.github.io/styleguide/jsguide.html#features-objects-enums

**Comment by yachtcaptain23** [comment]
Would you change the font color for CONFIRMED, INSTALLED, DOWNLOADED to blue to indicate it's clickable?

File: app/controllers/publishers/promo_registrations_controller.rb [hunk 2]
```diff
@@ -15,51 +49,99 @@ def overview
...
+  # groups       - the group information from Eyeshade
+  # user         - the publisher to find their finalizations
+  # start_date   - the beginning of the range to recieve data within
+  # end_date     - the end of the range to recieve data within
+  #
+  # Returns a hash of group_ids and their associated counts, { group_id => count }
+  def group_counts(groups, user, start_date, end_date)
+    # Create a { country => group_id } mapping hash
+    countries = groups.each_with_object({}) do |group, hash|
+      group[:codes].each { |code| hash[code] = group[:id] }
```
Comment (yachtcaptain23) [inline app/controllers/publishers/promo_registrations_controller.rb]: Is this part merged in yet to production?

Comment (corymcdonald) [reply app/controllers/publishers/promo_registrations_controller.rb]: It's the section `PromoClient.reporting.geo_stats_by_referral_code`. It returns strictly countries right now but what we want is country codes. 🌐

**Comment by corymcdonald** [comment]
@yachtcaptain23 I updated the component to make it look a little more visually selectable rather than just blue link text because it was hard to distinguish even afterwards.

Dec-16-2019 15-49-34
Dec-16-2019 15-50-30

File: app/javascript/packs/referrals/ReferralPanel.tsx (hunk 1 above)
Comment (corymcdonald) [reply app/javascript/packs/referrals/ReferralPanel.tsx]: https://github.com/brave-intl/publishers/pull/2428/commits/733cfdf408e81200e312f8bbc9f4f6063bc4027f

**Pull request merged**